lib/ask/runtime/questionnaire_ivr_simulator.ex

```diff
@@ -125,6 +121,25 @@ defmodule Ask.Runtime.QuestionnaireIvrSimulator do
       sync_simulation
     )
   end
+
+  defp append_ao_message(simulation, nil) do
+    simulation
+  end
+
+  defp append_ao_message(simulation, reply) do
+    messages = simulation.messages ++ AOMessage.create_all(reply)
+    Map.put(simulation, :messages, messages)
+  end
+
+  defp handle_at_message(simulation, "timeout") do
```

Comment:
For the backend `message` is a `String.t()`, hence it can either be one or more digits (0-9) or a special message (e.g. timeout), and it looks like the proper way to handle the situation in Erlang/Elixir, and it helped reduce the complexity in the calling functions.